File: solutions/2026-08-11/java/2026-08-11-smallest-missing-integer-greater-than-sequential-prefix-sum.java
```java
// Problem: Find the smallest missing integer greater than or equal to the sum of the longest sequential prefix.
// Link: https://leetcode.com/problems/smallest-missing-integer-greater-than-sequential-prefix-sum/
// Approach:
// 1. Find the longest sequential prefix. Iterate through the array and check if nums[i] is equal to nums[i-1] + 1.
//    The prefix ends when this condition is no longer met or when the end of the array is reached.
// 2. Calculate the sum of this longest sequential prefix.
// 3. To efficiently check for missing integers, convert the input array `nums` into a HashSet.
// 4. Start checking from the calculated prefix sum. Increment the sum until a number is found that is not present in the HashSet.
//    This number will be the smallest missing integer greater than or equal to the prefix sum.
// Time Complexity: O(N) for finding the prefix and calculating the sum, O(N) for populating the HashSet, and O(M) in the worst case for finding the missing number, where M is the smallest missing number. Since N and nums[i] are small (<= 50), M will also be relatively small. Thus, the overall time complexity is dominated by O(N).
// Space Complexity: O(N) to store the elements of `nums` in a HashSet.

import java.util.HashSet;
import java.util.Set;

class Solution {
    public int findSmallestMissingInteger(int[] nums) {
        int n = nums.length;
        int prefixSum = 0;
        int longestSequentialPrefixLength = 0;

        // 1. Find the longest sequential prefix and its sum.
        // The first element always forms a sequential prefix of length 1.
        prefixSum = nums[0];
        longestSequentialPrefixLength = 1;

        // Iterate from the second element to find the end of the sequential prefix.
        for (int i = 1; i < n; i++) {
            // Check if the current element continues the sequential pattern.
            if (nums[i] == nums[i - 1] + 1) {
                prefixSum += nums[i]; // Add to the sum if it's part of the sequential prefix.
                longestSequentialPrefixLength++; // Extend the length of the sequential prefix.
            } else {
                // The sequential prefix is broken, so we stop.
                break;
            }
        }

        // 2. Store all numbers from the input array in a HashSet for efficient lookups.
        Set<Integer> numSet = new HashSet<>();
        for (int num : nums) {
            numSet.add(num);
        }

        // 3. Start checking from the prefix sum.
        int currentNumber = prefixSum;
        while (true) {
            // If the current number is not found in the set, it's our smallest missing integer.
            if (!numSet.contains(currentNumber)) {
                return currentNumber;
            }
            // If the current number is in the set, check the next integer.
            currentNumber++;
        }
    }
}
```

File: solutions/2026-08-11/cpp/2026-08-11-smallest-missing-integer-greater-than-sequential-prefix-sum.cpp
// Problem: Smallest Missing Integer Greater Than Sequential Prefix Sum
// Link: https://leetcode.com/problems/smallest-missing-integer-greater-than-sequential-prefix-sum/
//
// Approach:
// First, we need to find the longest sequential prefix. We can iterate through the array,
// keeping track of the current sum and checking if the next element is exactly one greater
// than the previous element. If the sequence breaks, we stop and record the sum of the
// sequential prefix found so far.
// After finding the sum of the longest sequential prefix, we need to find the smallest
// integer greater than or equal to this sum that is missing from the array.
// A simple way to do this is to use a hash set (unordered_set in C++) to store all elements
// of the input array for quick lookups. Then, we can start checking from the sum of the
// sequential prefix and increment until we find a number that is not present in the hash set.
//
// Time Complexity:
// Finding the longest sequential prefix takes O(N) time, where N is the length of nums.
// Inserting all elements into the hash set takes O(N) time.
// Checking for missing integers starts from the prefix sum and increments. In the worst case,
// we might check up to N+1 numbers (if the prefix sum is small and all numbers up to that point are present).
// So, checking for the missing integer takes O(N) time in the worst case.
// Therefore, the overall time complexity is O(N).
//
// Space Complexity:
// We use a hash set to store all elements of nums, which takes O(N) space.
// The variables used for calculation take O(1) space.
// Therefore, the overall space complexity is O(N).

#include <vector>
#include <numeric>
#include <unordered_set>

class Solution {
public:
    int findSmallestInteger(std::vector<int>& nums) {
        // Calculate the sum of the longest sequential prefix.
        long long current_sum = 0; // Use long long to avoid potential overflow for sum.
        int n = nums.size();

        // Handle the case where the first element itself starts a sequential prefix.
        if (n > 0) {
            current_sum = nums[0]; // Initialize sum with the first element.
            // Iterate from the second element to find the longest sequential prefix.
            for (int i = 1; i < n; ++i) {
                // Check if the current element is exactly one greater than the previous element.
                if (nums[i] == nums[i - 1] + 1) {
                    current_sum += nums[i]; // Extend the sequential prefix.
                } else {
                    // The sequential prefix is broken. Stop here.
                    break;
                }
            }
        }
        // If nums is empty, current_sum remains 0, and we'll start checking from 0.
        // However, the problem constraints state 1 <= nums.length <= 50, so nums is never empty.

        // Create a hash set to efficiently check for the presence of numbers.
        std::unordered_set<int> num_set;
        for (int num : nums) {
            num_set.insert(num);
        }

        // Start checking from the sum of the longest sequential prefix.
        // Find the smallest integer >= current_sum that is not in the set.
        long long missing_candidate = current_sum;
        while (true) {
            // Check if the current candidate is present in the set.
            if (num_set.find(static_cast<int>(missing_candidate)) == num_set.end()) {
                // If not found, this is our smallest missing integer.
                return static_cast<int>(missing_candidate);
            }
            // If found, increment the candidate and check again.
            missing_candidate++;
        }
    }
};
```